rero_ils/modules/api.py

```diff
@@ -137,7 +138,12 @@ def create(cls, data, id_=None, delete_pid=False,
         if not id_:
             id_ = uuid4()
         cls.minter(id_, data)
-        record = super(IlsRecord, cls).create(data=data, id_=id_, **kwargs)
+        # We don't need to keep the PID if any validation error
+        try:
+            record = super(IlsRecord, cls).create(data=data, id_=id_, **kwargs)
+        except (RecordValidationError, ValidationError) as e:
+            db.session.rollback()
```

Comment:
I'am not sure this is atomic. I think we have to use `with db.session.begin_nested():`

Reply:
I think so too. But then I check https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L172-L190 which already uses `db.session.begin_nested()`.

That's why I just use `db.session.rollback()`.

Reply:
Minter add the pid to the session here: https://github.com/inveniosoftware/invenio-pidstore/blob/master/invenio_pidstore/models.py#L562
The `with` statement should commit at the end. So PID should be reserved.

But I didn't notice that the PID remains after the rollback yet. I should investigate more.